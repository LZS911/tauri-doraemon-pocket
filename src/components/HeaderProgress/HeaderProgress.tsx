import { IHeaderProgressProps } from '.';
import LinearProgress from '@material/react-linear-progress';

const HeaderProgress: React.FC<IHeaderProgressProps<HTMLDivElement>> = ({
  className = '',
}) => {
  return (
    <LinearProgress
      bufferingDots={false}
      indeterminate={true}
      className={`absolute top-0 z-[9999999] ${className}`}
    />
  );
};

export default HeaderProgress;
