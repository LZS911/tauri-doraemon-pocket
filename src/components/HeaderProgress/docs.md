# Generate HeaderProgress component by gl-cli
