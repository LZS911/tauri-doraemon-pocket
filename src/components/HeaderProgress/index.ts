import { LinearProgressProps } from '@material/react-linear-progress';
import HeaderProgress from './HeaderProgress';
export interface IHeaderProgressProps<T> extends LinearProgressProps<T> {
  children?: React.ReactNode;
  className?: string;
}
export default HeaderProgress;
