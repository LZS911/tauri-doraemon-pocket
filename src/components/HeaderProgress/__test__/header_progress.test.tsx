import HeaderProgress from '..';
import { render } from '@testing-library/react';
describe('test HeaderProgress', () => {
  it('should render HeaderProgress to match snapshot', () => {
    const { container } = render(<HeaderProgress />);
    expect(container).toMatchSnapshot();
  });
});
