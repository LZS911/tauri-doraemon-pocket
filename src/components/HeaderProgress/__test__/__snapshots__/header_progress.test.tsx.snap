// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test HeaderProgress should render HeaderProgress to match snapshot 1`] = `
<div>
  <div
    class="mdc-linear-progress mdc-linear-progress--indeterminate absolute top-0 z-[9999999]  mdc-linear-progress--indeterminate"
    role="progressbar"
  >
    <div
      class="mdc-linear-progress__buffer"
      style="transform: scaleX(1);"
    />
    <div
      class="mdc-linear-progress__bar mdc-linear-progress__primary-bar"
      style="transform: scaleX(1);"
    >
      <span
        class="mdc-linear-progress__bar-inner"
      />
    </div>
    <div
      class="mdc-linear-progress__bar mdc-linear-progress__secondary-bar"
    >
      <span
        class="mdc-linear-progress__bar-inner"
      />
    </div>
  </div>
</div>
`;
